src/goto-programs/goto_program_template.h

```diff
@@ -386,21 +386,12 @@ class goto_program_templatet
     return t;
   }
 
-  static const irep_idt get_function_id(
-    const_targett l)
-  {
-    while(!l->is_end_function())
-      ++l;
-
-    return l->function;
-  }
-
   static const irep_idt get_function_id(
     const goto_program_templatet<codeT, guardT> &p)
   {
-    assert(!p.empty());
+    PRECONDITION(!p.empty());
 
-    return get_function_id(--p.instructions.end());
+    return p.instructions.front().function;
```

Comment:
As far as I recall, an important reason for grabbing the function name from the `END_FUNCTION` instruction was to work around the change of names that inlining might yield. But then that's all a bit messy so I wouldn't oppose the proposed change.